src/Scratch.as

```diff
@@ -1136,11 +1140,13 @@ public class Scratch extends Sprite {
 		function clearProject():void {
 			startNewProject('', '');
 			setProjectName('Untitled');
+			var clientID:String = offlineSystem.getClientId();
+			var appOpenPacket:Packet = new Packet(clientID, "Untitled", 'project', 'create', offlineSystem.getCurrentLanguage(), null);
+			offlineSystem.enqueuePacket(appOpenPacket);
```

Comment:
As far as I can tell this is the only tricky part of moving the `offlinesystem` variable into `ScratchOffline` in the `scratch-flash-offline` repository. I would recommend replacing these three lines with a call to, say, `onNewProject()`. In the `Scratch` class, you can just make an empty protected method. Then, in `ScratchOffline`, make an override for `onNewProject()` which contains these three lines.

That would keep the offline telemetry system contained within the `scratch-flash-offline` repository, and the only changes in `scratch-flash` would be supporting-but-generic changes.

Let me know if there's a reason that this won't work; maybe I can help come up with a solution.